src/H5Dmpio.c

```diff
@@ -808,10 +808,6 @@ H5D__chunk_collective_io(H5D_io_info_t *io_info, const H5D_type_info_t *type_inf
     HDassert(type_info);
     HDassert(fm);
 
-    /* Disable collective metadata reads for chunked dataset I/O operations
-     * in order to prevent potential hangs */
-    H5CX_set_coll_metadata_read(FALSE);
```

Comment:
This should no longer be necessary after these changes. The disabling of collective metadata reads is on a more granular scale rather than taking a hammer to it here.